src/app/core/services/scheduling.service.ts

```diff
@@ -405,36 +363,38 @@ export class SchedulingService {
 
   buildReportSchedule() {
     let iterDate = this.setDateTimeToMidnight(new Date(this.enrolmentDate))
-    const yearsMillis = DefaultScheduleYearCoverage * 60000 * 60 * 24 * 365
+    const yearsMillis = getMilliseconds({ years: DefaultScheduleYearCoverage })
     const endDate = new Date(iterDate.getTime() + yearsMillis)
     const tmpSchedule: ReportScheduling[] = []
 
     while (iterDate.getTime() <= endDate.getTime()) {
-      iterDate = this.advanceRepeat(
-        iterDate,
-        'day',
-        DefaultScheduleReportRepeat
+      iterDate = SchedulingService.advanceRepeat(iterDate, {
+        unit: 'day',
+        amount: DefaultScheduleReportRepeat
+      })
+      const report = SchedulingService.reportBuilder(
+        tmpSchedule.length,
+        iterDate
       )
-      const report = this.reportBuilder(tmpSchedule.length, iterDate)
       tmpSchedule.push(report)
     }
     console.log('[√] Updated report schedule.')
     return Promise.resolve(tmpSchedule)
   }
 
-  reportBuilder(index, reportDate): ReportScheduling {
-    const report = {
+  static reportBuilder(index: number, reportDate: Date): ReportScheduling {
+    const timestamp = reportDate.getTime()
+    return {
       index: index,
-      timestamp: reportDate.getTime(),
+      timestamp: timestamp,
       viewed: false,
       firstViewedOn: 0,
       range: {
         timestampStart:
-          reportDate.getTime() - DefaultScheduleReportRepeat * 60000 * 60 * 24,
-        timestampEnd: reportDate.getTime()
+          timestamp - DefaultScheduleReportRepeat * TIME_UNIT_MILLIS.day,
```

Comment:
I would use the time calc function. That makes it more explicit that we are converting days to milliseconds